File: src/Gostio.Services/Database/Entities/Message.cs
namespace Gostio.Services.Database.Entities;

public class Message
{
    public int Id { get; set; }

    public int ConversationId { get; set; }

    public Conversation Conversation { get; set; } = null!;

    public int SenderUserId { get; set; }

    public User SenderUser { get; set; } = null!;

    public string Body { get; set; } = null!;

    public DateTime SentAt { get; set; }
}

File: packages/gostio_core/lib/src/models/message.dart
import 'package:json_annotation/json_annotation.dart';

part 'message.g.dart';

@JsonSerializable(createToJson: false)
class Message {
  const Message({
    required this.id,
    required this.conversationId,
    required this.senderUserId,
    required this.senderName,
    required this.body,
    required this.sentAt,
  });

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);

  final int id;
  final int conversationId;
  final int senderUserId;
  final String senderName;
  final String body;
  final DateTime sentAt;
}
